packages/markup-frontstage/src/services/SavedView.ts

```diff
@@ -71,9 +80,72 @@ export const parseFromEmphasizeElements = (
 };
 
 /**
- * Creates saved view data
+ * Creates a drawing saved view.
+ * @param vp the name of the view port.
+ */
+const createDrawingSavedView = (vp: Viewport): SavedViewData => {
+  const viewState = vp.view as DrawingViewState;
+  if (!viewState) {
+    throw new Error("Invalid viewport");
+  }
+
+  const categorySelectorProps = viewState.categorySelector.toJSON();
```

Comment:
maybe ensure that categorySelector and displayStyle are defined?

Reply:
ensured please!